### ISSUE TITLE
Issue with media control buttons on Windows 10

### ISSUE THREAD

**Issue by eera5607** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#3598]
Hello. I know that very similar issues had been published before but I haven't found any working solution. I don't know if this is very specific to my case but I think it might not be. 

**Issue Descriptions:**

Media control buttons (Play/Pause, Next, Previous) on Logitech K780 using Logitech Options software don't work as expected. If you activate _Show track information in Windows 10_ volume overlay all buttons work, the issue is that next and previous keys jump 2 tracks instead of one. If I disable _Show track information in Windows 10_ option, media keys don't work at all. I also have a Logitech MX Anywhere 2S mouse which have keys that you can configure to work as media keys and in this case, even with the option Show track information in Windows 10 disabled they work normally to play/pause and change tracks. GPMDP is getting different signals from each device but if you activate _Show track information in Windows 10_ even the mouse previous and next keys jump 2 tracks instead of one. This is all with the Playback API activated and JSON API disabled. If I left JSON enable media keys work on the mouse but no the keyboard. I don't know if there is already a solution that works with Logitech Options but I can't find it. I found a solution for Logitech SetPoint but is not applicable in this case. 

Thanks.

**OS:**

Windows 10 1903

**GPMDP Version:**

4.6.1

**Comment by stefanjm** [comment]
Using media next/previous buttons on my acer laptop keyboard also causes it to jump over 1 track.

**Comment by gcfrxbots** [comment]
[MediaKeyFix.zip](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/files/3513592/MediaKeyFix.zip)

I made an AHK file to somewhat fix this. It'll keep the play/pause button from doublepressing (and thus making it useless) and will make the 'previous track' button work, however due to AHK simulating the keypresses I'm unable to get the menu to appear. The skip track will still double skip, but if you see a song you want then you can go back easily.

Is there any ETA on a fix for this?

**Comment by schaffrath94** [comment]
I have the same issue. Its very annoying... Any sign of fixing the problem?
I use the Corsair K70.

**Comment by eVen-gits** [comment]
Just to add to this:

It's also not working in win8.1 or on linux (kubuntu 18.04 LTS).

**Comment by RubixSolver99** [comment]
I am having the same issue, I cannot use the media keys on my keyboard or my programmed macros on my mouse, without double-pressing the skip/prev/play/pause buttons.

**Comment by IndefiniteBen** [comment]
Previously reported multiple times in #2364, #3293 and MANY other issues. At this point we're basically waiting for @Kytech to send in a pull request for the change that fixes this? I haven't looked into it in a while though...

**Comment by Kytech** [comment]
@IndefiniteBen I've been fighting the electron build process the last couple days to allow for GPMDP to build on my local machine. Ever since GPMDP updated to Electron 3, I've had build issues that I just recently resolved. I've managed to fix the double-skipping bug, but the play/pause issue is a new one that has recently come up. I'll have to do some more digging into the play/pause issue. I'm thinking that it is also related to the issue of the media information not actually displaying in the overlay.

**Comment by 1024mb** [comment]
Still present with latest version. I think, iirc, that it was fixed for a couple of days but I'm not sure.

**Comment by Kytech** [comment]
@1024mb, we're currently waiting on some tests to be done for the pull request that I submitted is accepted since @MarshallOfSound said he wants to test my fix further before merging. @MarshallOfSound, if there are any tests that I could help with, I'd love to look into it. It seems like enough people are experiencing the issue that I'd be more than willing to try and help speed things up if I can.

**Comment by 1024mb** [comment]
> 
> 
> @1024mb, we're currently waiting on some tests to be done for the pull request that I submitted is accepted since @MarshallOfSound said he wants to test my fix further before merging. @MarshallOfSound, if there are any tests that I could help with, I'd love to look into it. It seems like enough people are experiencing the issue that I'd be more than willing to try and help speed things up if I can. In the meantime, you can download my fix from the AppVeyor build [at this link.](https://ci.appveyor.com/api/buildjobs/e908ps71t93fc0wx/artifacts/dist%2Finstallers%2Fwin32%2FGoogle%20Play%20Music%20Desktop%20PlayerSetup.exe)

Thanks for your build but I got a message Youtube Music isn't optimized for this browser and that I should download Chrome.

**Comment by Kytech** [comment]
YouTube Music support is still more experimental, but I think it should still have basic functionality. You should be able to ignore that message and use YTMusic anyway. If not, go ahead and open another issue with as much detail as you can give.

**Comment by Kytech** [comment]
> > @1024mb, we're currently waiting on some tests to be done for the pull request that I submitted is accepted since @MarshallOfSound said he wants to test my fix further before merging. @MarshallOfSound, if there are any tests that I could help with, I'd love to look into it. It seems like enough people are experiencing the issue that I'd be more than willing to try and help speed things up if I can. In the meantime, you can download my fix from the AppVeyor build [at this link.](https://ci.appveyor.com/api/buildjobs/e908ps71t93fc0wx/artifacts/dist%2Finstallers%2Fwin32%2FGoogle%20Play%20Music%20Desktop%20PlayerSetup.exe)
> 
> Thanks for your build but I got a message Youtube Music isn't optimized for this browser and that I should download Chrome.

YouTube Music support is still more experimental, but I think it should still have basic functionality. You should be able to ignore that message and use YTMusic anyway. If not, go ahead and open another issue with as much detail as you can give.

**Comment by RacingJayson** [comment]
> @1024mb, we're currently waiting on some tests to be done for the pull request that I submitted is accepted since @MarshallOfSound said he wants to test my fix further before merging. @MarshallOfSound, if there are any tests that I could help with, I'd love to look into it. It seems like enough people are experiencing the issue that I'd be more than willing to try and help speed things up if I can. In the meantime, you can download my fix from the AppVeyor build [at this link.](https://ci.appveyor.com/api/buildjobs/e908ps71t93fc0wx/artifacts/dist%2Finstallers%2Fwin32%2FGoogle%20Play%20Music%20Desktop%20PlayerSetup.exe)

Your patched version worked and fixed the issue with me! Thank you

**Comment by sli** [comment]
I'm having this issue and it is absolutely not related to any specific keyboard because it happens on my fully custom QMK board, my small macropad, AND the media controls on my headphones. Tested it with other media players and they do not have the same issue, so it's got to be a GPMDP bug.
